File: vampire.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main() {
    int initialPower,n,t,result,greenSerums,blueSerums;
    cin>>t;
    for(int i=0;i<t;i++){
        vector<int> welderPowers;
        greenSerums = 2;
        blueSerums = 1;
        cin>>n>>initialPower;
        for(int j=0;j<n;j++){
            int temp;
            cin>>temp;
            welderPowers.push_back(temp);
        }
        result = n;
        sort(welderPowers.begin(), welderPowers.end());
        for (int j = 0; j < n; j++) {
            if (initialPower > welderPowers[j]) {
                initialPower += (welderPowers[j] / 2);
            } else {
                if (greenSerums > 0 && 2 * initialPower > welderPowers[j]) {
                    initialPower *= 2;
                    greenSerums -= 1;
                    initialPower += (welderPowers[j] / 2);
                } else if (blueSerums > 0 && 3 * initialPower > welderPowers[j]) {
                    initialPower *= 3;
                    blueSerums -= 1;
                    initialPower += (welderPowers[j] / 2);
                } else if (greenSerums > 1 && 4 * initialPower > welderPowers[j]) {
                    initialPower *= 4;
                    greenSerums -= 2;
                    initialPower += (welderPowers[j] / 2);
                } else {
                    result = j;
                    break;
                }
            }
        }
        cout<<result<<endl;
    }
    return 0;
}

File: vampire.py
def vampire(initialPower,welderPowers):
    n=len(welderPowers)
    welderPowers.sort()
    greenSerums=2
    blueSerums=1
    for i in range(n):
        if initialPower> welderPowers[i]:
            initialPower+=(welderPowers[i]//2)
        else:
            if greenSerums>0 and 2*initialPower>welderPowers[i]:
                initialPower*=2
                greenSerums-=1
                initialPower+=(welderPowers[i]//2)
            elif blueSerums>0 and 3*initialPower>welderPowers[i]:
                initialPower*=3
                blueSerums-=1
                initialPower+=(welderPowers[i]//2)
            elif greenSerums>1 and 4*initialPower>welderPowers[i]:
                initialPower*=4
                greenSerums-=2
                initialPower+=(welderPowers[i]//2)
            else:
                return i+1
    return n
print(vampire(1,[2,1,8,9]))
print(vampire(3,[6,2,60]))